src/autogluon_dashboard/scripts/widget.py

```diff
@@ -4,6 +4,63 @@
 
 
 class Widget:
+    """
+    This class is used to define any Panel widgets that are used on the dashboard website.
+    These widgets can be independent or attached to a dashboard to control the plot.
+    ...
+
+    Attributes
+    ----------
+    widget_type: str
+        the widget type is used to decide what type of panel widget will be created [refer to the if-else statements in __init__]
+    name: str
+        name of the widget on the website
+    value: (int/str)
+        for applicable widgets, this is the value that will be displayed [refer to number and slider widgets]
+    options: (str/List[str])
+        for applicable widgets, this is the list of options that will be displayed [refer to select and toggle widgets]
+    format: str
+        for applicable widgets, this is the format of the "value" being displayed [refer to the number indicator widget]
+    start: int
+        for applicable widgets, this is the start of a range being displayed [refer to the slider widget]
+    end: int
+        for applicable widgets, this is the end of a range being displayed [refer to the slider widget]
+    file: str
+        for applicable widgets, this is a file object to upload/download
+    filename: str
+        for applicable widgets, this is the name of the file to upload/download
+
+    Methods
+    -------
+    _create_selectwidget():
+        Creates a panel `Select` widget.
+        The `Select` widget allows selecting a `value` from a list of `options`.
+
+    _create_togglewidget():
+        Creates a panel `ToggleGroup` widget.
+        The `ToggleGroup` is a group of widgets which can be switched 'on' or 'off'. It can be displayed as buttons or boxes (default).
+
+    _create_numberwidget():
+        Creates a panel `Number` indicator.
+        The `Number` indicator renders the `value` as text.
+
+    _create_sliderwidget():
+        Creates a panel `IntSlider` widget.
+        The `IntSlider` widget allows selecting an integer value within a set of bounds using a slider.
+
+    _create_downloadwidget():
+        Creates a panel `FileDownload` widget.
+        The `FileDownload` widget allows a user to download a file.
+        It works either by sending the file data to the browser on initialization (`embed`=True), or when the button is clicked.
+    ...
+
+    Usage
+    -------
+    widget = Widget(widget_type=<widget_type>, name=<widget_name>).create_widget()
```

Comment:
Something might be useful to learn for you is the factory design pattern. Where a factory is going to craft the object, and the client doesn't really care what the concrete object type is. They just know that it's a `Widget`, then they use the standard interface to interact with all the `Widgets`, i.e. `widget.create`

Reply:
Oh I see, so instead of using sub "functions" we use a sub class (`SliderWidget` instead of `_create_selectwidget`)